### PR TITLE
[sercomp] Add experimental `--quick` option to skip proofs.

### PR DESCRIPTION
**Pull request by ejgallego**
IMHO this is not an option we want to support in the long term as
proposed here, however @palmskog proposed it and indeed it may be very
useful in the short term.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ _Version 0.6.0_:
              (@ejgallego, report by @palmskog)
  * [sercomp] Add `check` mode to compilers to check all proofs without
              outputting `.vo`. (@palmskog)
+ * [sercomp] Add "hacky" `--quick` mode to skip opaque proofs.
```
Comment (palmskog) [inline CHANGES.md]: Isn't this an "option" rather than a "mode"? Or do we want to keep that distinction?

Comment (ejgallego) [reply CHANGES.md]: Indeed that's more of an option, as it does apply to several modes; thanks, will amend. I guess I was influenced subconsciously by the internal Coq option being named `async_proofs_mode` :) 

IMHO we would want a bit of a different API for proof checking in general, so parsing would never check proofs by default, but for now we are forced to expose the internal Coq modes and a bit of a combinatorial mess.

**Comment by ejgallego** [comment]
> For example, the following might be enough to keep it interesting:

Thanks for the review. I actually like the large test case as it turns out to stress a couple of things; and of course I've added the one you suggested.

**Comment by ejgallego** [comment]
Actually, _a coup d'oeil_, in order to test this functionality properly we would have more than 100 different cases based on the code of the STM, so yeah, we can only hope to add what we find by now.

**Pull request merged**